middle_end/flambda2/to_cmm/to_cmm_static.ml

```diff
@@ -231,7 +231,31 @@ let static_const0 env res ~updates (bound_static : Bound_static.Pattern.t)
     in
     let static_fields = List.concat_map (static_field res) fields in
     let block = C.emit_block sym header static_fields in
-    let env, res, updates = static_block_updates sym env res updates 0 fields in
+    let update_kinds =
+      match shape with
+      | Value_only -> List.map (fun _ -> UK.pointers) fields
+      | Mixed_record shape ->
+        let value_prefix =
+          List.init (Flambda_kind.Mixed_block_shape.value_prefix_size shape)
+            (fun _ -> UK.pointers)
+        in
+        let flat_suffix =
+          List.map
+            (fun (flat_suffix_elt : Flambda_kind.flat_suffix_element) ->
+              match flat_suffix_elt with
```

Comment:
this is at least the second occurrence of this function (the first being in the recently merged #2997 ). This should be factored out somewhere, either in this file, or provided in the `Mixed_block_shape` module.